packages/desktop-client/src/components/reports/CategorySelector.tsx

```diff
@@ -24,23 +26,72 @@ export default function CategorySelector({
 }: CategorySelectorProps) {
   const [uncheckedHidden, setUncheckedHidden] = useState(false);
 
+  const allCategoriesSelected = categoryGroups.every(categoryGroup =>
+    categoryGroup.categories.every(category =>
+      selectedCategories.some(
+        selectedCategory => selectedCategory.id === category.id,
+      ),
+    ),
+  );
+
+  const allCategoriesUnselected = !categoryGroups.some(categoryGroup =>
+    categoryGroup.categories.some(category =>
+      selectedCategories.some(
+        selectedCategory => selectedCategory.id !== category.id,
+      ),
+    ),
+  );
+
+  const selectAll = () => {
+    let test = [];
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏poor variable naming